### PR TITLE
Show realm migration progress via a simple UI

### LINKED ISSUE #16554: A UI needs to be displayed during realm database migration
It can take a while to run, and it's currently not shown anywhere except in a (usually invisible) console.

Already PR'd but creating a tracking issue.

### PR DESCRIPTION
**Pull request by peppy**
This is working and ready to go, but needs a touch more consideration with regards to subscriptions. Will be addressing that separately and blocking this for now.

- Closes https://github.com/ppy/osu/issues/16554

### CODE REVIEW

**Comment by peppy** [comment]
Unblocking, added dependency on subscription fixes PR.

File: osu.Game/Database/EFToRealmMigrator.cs [hunk 1]
```diff
@@ -4,52 +4,130 @@
...
+            };
         }
 
-        public void Run()
+        protected override void LoadComplete()
         {
+            base.LoadComplete();
+
+            // needs to be run on the update thread because of realm BlockAllOperations.
+            // maybe we can work around this? not sure..
```
Comment (frenzibyte) [inline osu.Game/Database/EFToRealmMigrator.cs:104]: Is this comment still valid? this is invoked in `LoadComplete` which should guarantee running from update thread.

**Comment by peppy** [comment]
In the interest of keeping things moving forward as I have a lot more realm changes incoming, I've fixed this to not require the subscription changes. As pointed out by my own comment, the only reason we're facing issues in the first place is due to taking the backup. So I've moved that backup to `OsuGameBase` to have it run before other realm usages.

Note that this means running visual tests will take a backup but not migrate. I don't see that as a huge issue though - it will resolve the first time each dev runs the game.

**Comment by bdach** [comment]
I've fixed a code inspection and resolved merge conflicts here. Please let me know if the resolution looks correct.

Otherwise this looks pretty good. One minor detail that it may be good to fix is to move the following log statement:

https://github.com/ppy/osu/blob/93a2a2df35bc88edd4c9db0aabf46f60038307d7/osu.Game/Database/DatabaseContextFactory.cs#L154-L155

out of `DatabaseContextFactory.Backup()` and into the `EFToRealmMigrator` screen, especially so that the two aren't really tied together that tightly (backups run in base game after all). But given that's a dev-only log statement then maybe fine to leave?

**Comment by peppy** [comment]
Resolution looks good. Up to you whether you move the log, I don't think it matters too much (it's at the point of the EF database being moved out, so I think it makes sense where it is?)

**Comment by bdach** [comment]
> Up to you whether you move the log

Have applied 7e68371.

**Review by bdach — APPROVED** [review]

**Pull request merged**